### PR TITLE
SONAR-6646 Move computation of issues and new issues measures to Compute Engine

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

**Comment by sns-seb** [comment]
some bug to fix, PR is yours

**Comment by julienlancelot** [comment]
@seb feedback applied !

**Comment by julienlancelot** [comment]
Last feedback apply

**Comment by sns-seb** [comment]
LGTM

but if you could refactor your unit test to use MeasureAssert, that would be great

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeIssueMeasuresStep.java [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+
+  private final static Map<String, String> SEVERITY_METRIC_KEY_BY_SEVERITY = ImmutableMap.of(
+    Severity.BLOCKER, BLOCKER_VIOLATIONS_KEY,
+    Severity.CRITICAL, CRITICAL_VIOLATIONS_KEY,
+    Severity.MAJOR, MAJOR_VIOLATIONS_KEY,
+    Severity.MINOR, MINOR_VIOLATIONS_KEY,
+    Severity.INFO, INFO_VIOLATIONS_KEY
+  );
+
+  private final static Map<String, String> NEW_SEVERITY_METRIC_KEY_BY_SEVERITY = ImmutableMap.of(
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeIssueMeasuresStep.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeIssueMeasuresStep.java [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+      }
+      return true;
+    }
+
+    public int getSum() {
+      return sum;
+    }
+
+    private static boolean isAfter(Issue issue, long date) {
+      // TODO should we truncate the date to the second as it was done in batch ?
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeIssueMeasuresStep.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeIssueMeasuresStep.java [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+ */
+public class ComputeIssueMeasuresStep implements ComputationStep {
+
+  private final BatchReportReader reportReader;
+  private final TreeRootHolder treeRootHolder;
+  private final PeriodsHolder periodsHolder;
+  private final MeasureRepository measureRepository;
+  private final MetricRepository metricRepository;
+
+  private final static Map<String, String> SEVERITY_METRIC_KEY_BY_SEVERITY = ImmutableMap.of(
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeIssueMeasuresStep.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MINOR 2 minor
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by julienlancelot** [comment]
@simonbrandhof Please merge this PR on master when goldeneye will be green

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeIssueMeasuresStep.java [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+    Severity.BLOCKER, BLOCKER_VIOLATIONS_KEY,
+    Severity.CRITICAL, CRITICAL_VIOLATIONS_KEY,
+    Severity.MAJOR, MAJOR_VIOLATIONS_KEY,
+    Severity.MINOR, MINOR_VIOLATIONS_KEY,
+    Severity.INFO, INFO_VIOLATIONS_KEY
+    );
+
+  private final static Map<String, String> NEW_SEVERITY_METRIC_KEY_BY_SEVERITY = ImmutableMap.of(
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeIssueMeasuresStep.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 new issues:
* 3 major
* 2 minor
* 1 info

Details:
* Add a 'package-info.java' file to document the 'src/main/java/org/sonar/batch/bootstrap' package (org.codehaus.sonar:sonar-batch:src/main/java/org/sonar/batch/bootstrap) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* Add a 'package-info.java' file to document the 'src/main/java/org/sonar/server/computation/measure' package (org.codehaus.sonar:sonar-server:src/main/java/org/sonar/server/computation/measure) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* Add a 'package-info.java' file to document the 'src/main/java/org/sonar/server/computation/step' package (org.codehaus.sonar:sonar-server:src/main/java/org/sonar/server/computation/step) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)

**Pull request merged**